src/app/clusters/access-control-server/access-control-server.cpp

```diff
@@ -489,17 +514,20 @@ CHIP_ERROR AccessControlAttribute::ReadExtension(AttributeValueEncoder & aEncode
     auto & fabrics = Server::GetInstance().GetFabricTable();
 
     return aEncoder.EncodeList([&](const auto & encoder) -> CHIP_ERROR {
-        for (auto it = fabrics.begin(); it != fabrics.end(); ++it)
+        for (auto & fabric : fabrics)
         {
             uint8_t buffer[kExtensionDataMaxLength] = { 0 };
             uint16_t size                           = static_cast<uint16_t>(sizeof(buffer));
-            auto err =
-                storage.SyncGetKeyValue(key.AccessControlExtensionEntry(kStorageVersion, it->GetFabricIndex()), buffer, size);
-            ReturnErrorCodeIf(err == CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND, CHIP_NO_ERROR);
-            ReturnErrorOnFailure(err);
+            CHIP_ERROR errStorage = storage.SyncGetKeyValue(key.AccessControlExtensionEntry(fabric.GetFabricIndex()), buffer, size);
+            VerifyOrDie(errStorage != CHIP_ERROR_BUFFER_TOO_SMALL);
```

Comment:
VerifyOrDie is too strict here. We should skip or delete the extension on CHIP_ERROR_BUFFER_TOO_SMALL, which would be a definite internal error

Reply:
I thought reviewers might say that but thought I'd try it. It's definitely an internal error, we should never be writing too large a value, and if we ever bump it up, it'll be bumped in the buffer size too. But I can relax the handling.